### DISCUSSION TITLE
What python version is recomended for pyenv?

### DISCUSSION THREAD

**Discussion by UnconnectedBedna** [Stability-AI/StableSwarmUI#327] [Q&A]
As title.
Latest available (3.12.3 for me) or some older version?

I can not find any mention of python version anywhere.

**Comment by mcmonkey4eva** [comment]
- Swarm itself does not use python
- Python choice would be specific to [comfy](https://github.com/comfyanonymous/ComfyUI) rather than swarm
- When using swarm's installer for Comfy, it will set up the env for you (on windows at least) so you don't need to worry about it
- 3.10 is most intercompatible, 3.11 is stable (mostly), 3.12 is experimental (ie breaks a lot of common custom nodes)

**Comment by UnconnectedBedna** [comment]
So your answer is "there is no recommendation"?

I do not care for installers and such, I install everything manually on linux and I therefore need to know python recommendations.
Installing 3 versions of python in a pyenv to test is outside of the scope of what I am willing to do.

**Reply by mcmonkey4eva** [reply]
> 3.10 is most intercompatible, 3.11 is stable (mostly), 3.12 is experimental (ie breaks a lot of common custom nodes)

Pick whichever one of the three matches your case. Do you intend to use a lot of custom nodes n wotnot for comfy? If so, 3.10
Are you on a modern nvidia card and want stable/fast/best and don't expect to pile on weird nodes? 3.11
Do you like cutting edge stuff that works best but breaks easiest? 3.12 (you will have features broken *out of the box* to deal with iirc at least rembg is broken on 3.12)
